src/backend/distributed/commands/role.c

```diff
@@ -776,3 +788,141 @@ ConfigGenericNameCompare(const void *a, const void *b)
 	 */
 	return pg_strcasecmp(confa->name, confb->name);
 }
+
+
+/*
+ * SplitConfigurationList --- parse a string containing list of configuration values
+ *
+ * Most of the logic is copied from SplitIdentifierString that is the guts of
+ * textToQualifiedNameList with some changes:
+ * - Unquoted identifiers is not downcased here
+ * - The item seperator is fixed to be a comma as this is the case in pg_db_role_setting
+ * - Simplified set of parameters
+ * - Better error mechanisms
+ * - Replace banned memmove calls with pnstrdup
+ *
+ * Inputs:
+ *	configurationList: the input string; must be overwritable!	On return, it's been
+ *					   modified to contain the separated identifiers.
+ *
+ * Returns list of strings that are allocated in the configurationList string
+ *
+ * Note that an empty string is considered okay here, though not in
```

Comment:
Outdated comment. I will either add an assertion here, or remove this lines about parsing empty string